modules/objdetect/src/aruco/aruco_detector.cpp

```diff
@@ -1315,24 +1315,25 @@ void drawDetectedMarkers(InputOutputArray _image, InputArrayOfArrays _corners,
     int nMarkers = (int)_corners.total();
     for(int i = 0; i < nMarkers; i++) {
         Mat currentMarker = _corners.getMat(i);
-        CV_Assert(currentMarker.total() == 4 && currentMarker.type() == CV_32FC2);
+        CV_Assert(currentMarker.total() == 4 && currentMarker.channels() == 2);
+        currentMarker.convertTo(currentMarker, CV_32SC2);
```

Comment:
It makes sense, to check if the input is already `32SC2`. `convertTo` does not check it.

Reply:
added check:
```
        if (currentMarker.type() != CV_32SC2)
            currentMarker.convertTo(currentMarker, CV_32SC2);
```